sdk/core/azure-core/azure/core/_pipeline_client_async.py

```diff
@@ -168,3 +172,46 @@ def _build_pipeline(self, config, **kwargs): # pylint: disable=no-self-use
             transport = AioHttpTransport(**kwargs)
 
         return AsyncPipeline(transport, policies)
+
+    async def _make_pipeline_call(self, request, stream, **kwargs):
+        rest_request, request_to_run = _prepare_request(request)
+        return_pipeline_response = kwargs.pop("_return_pipeline_response", False)
+        pipeline_response = await self._pipeline.run(
+            request_to_run, stream=stream, **kwargs  # pylint: disable=protected-access
+        )
+        response = pipeline_response.http_response
+        if rest_request:
+            rest_response = response._to_rest_response()  # pylint: disable=protected-access
+            if not stream:
+                # in this case, the pipeline transport response already called .load_body(), so
+                # the body is loaded. instead of doing response.read(), going to set the body
+                # to the internal content
+                rest_response._content = response.body()  # pylint: disable=protected-access
+                await rest_response.close()
+            response = rest_response
+        if return_pipeline_response:
+            pipeline_response.http_response = response
+            pipeline_response.http_request = request
+            return pipeline_response
+        return response
+
+    def send_request(
```

Comment:
Not async?

Reply:
yeah, it's a little confusing, but it returns an awaitable object. This has to do with the fact that we're returning a context manager that both has an `__await__` and a `__aenter__` etc. So this way, you can call it like this:
```python
response = await client.send_request(request)
```
and like this (mainly in the case of streaming)
```python
async with client.send_request(request) as response:
    print(response)
```

talked through this with @johanste and he's ok with not making it `async def`. Reason is once we mark it as `async def`, the compiler thinks we're returning a coroutine, which don't have attributes `__aenter__`, so we get an attribute error when trying to use it as a context managed response